doc/design/model_market.md

```diff
@@ -0,0 +1,119 @@
+
+In the [model zoo design](model_zoo.md), we described how model developers develop, publish and share custom models on SQLFlow, and how analysts can make use of the shared model with SQLFlow.
+
+Here, we'll describe how to build the model market model developers and analysts.
+
+## Overview
+
+The model market is designed to:
+
+1. Manage user login and logout.
+1. View published images available for the current user.
+1. View published trained models available for the current user.
+1. Publish (or remove) model definition images.
+1. Publish (or remove) trained models.
+1. Share (or remove share) model definition images to other users.
+1. Share (or remove share) trained models to other users.
+
+<p align="center">
+<img src="figures/model_market_overview.png">
+</p>
+
+In order to support publishing and sharing models securely, the model market is able to communicate with some [SSO](https://en.wikipedia.org/wiki/Single_sign-on) service to authenticate users. Only users that are logged in can do operations on the model market.
+
+Then model definition images and trained model can have below accessibility settings:
+
+1. Private: Only visible to the current user.
+2. Public: Readable by every user.
+3. Private but shared to some users: visible to the current user and users that shared to.
+
+## Steps to View Model Definitions and Trained Models
+
+1. Login to model market.
+1. Click at the "Model Definitions" tab to see the list of model definition Docker images and the model class names in each Docker image.
+1. Click at "Trained Models" tab to see all trained models the current user have published by using SQLFlow `PUBLISH` statement, the evaluation result of the trained model will also be available.
+
+## Steps to Publish a Docker Image
+
+1. Login to model market.
+1. Go to the "Model Definitions" tab.
+1. Click "New" to add a new image.
+1. Input the full Docker image address in the dialog and click "OK".
+1. ***Optional***: The system will call the Docker registry API to grant access for SQLFlow to pull the image. When using a public image, this step will be skipped. If access can be granted, a message should be shown on the web page.
+1. The system will start to run several checks and tests using the Docker image. If all the checks have passed, the image is added.
+
+**NOTE: Docker image with different [tags](https://www.freecodecamp.org/news/an-introduction-to-docker-tags-9b5395636c2a/) will be recognized as different images.
+
+## Steps to Share Model Definition Docker Images to Other Users
+
+1. Login to model market.
+1. Go to the "Model Definitions" tab.
+1. Find a Docker image in the list to be shared and click on the button "share".
+1. Input the user name or user ID to share to and click "OK".
+
+The system will call Docker registry API to grant access for the user shared to after these operations. The model market will save the grant information in a database table `image_shares` that have below columns:
```

Comment:
Is it allowed for other users to download this Docker image, or this image can only be used in the current SQLFlow deployment?

Reply:
User's Docker image pull access is only granted to the SQLFlow account by adding this image into model market. Other users do not have access unless the current user share this image with him.

Reply:
If other users are granted access, they can download the image to their local machine?

Reply:
Yes, they can.